src/components/TodoTable.vue

```diff
@@ -1,127 +1,83 @@
 <script setup lang="ts">
-import axios from "axios"
-import { onMounted, ref } from "vue"
+import { computed, onMounted, ref } from "vue";
+import axios from "axios";
+import { RouterLink, useRouter } from "vue-router";
+import { deleteTodo } from '../apicalls'
+
+const router = useRouter();
 
 type TodoData = {
-  isComplete: boolean
-  todoName: string
-  createdAt: string
-  updatedAt: string
-  __v: number
-  _id: string
-}
+  isComplete: boolean;
+  todoName: string;
+  createdAt: string;
+  updatedAt: string;
+  __v: number;
+  _id: string;
+};
 
 type TodoResponse = {
-  code: number
-  data: TodoData[]
-}
-
-type TodoObject = Omit<TodoData, "createdAt" | "updatedAt" | "__v" | "_id">
-
-const title = ref("")
-const complete = ref(false)
-const allData = ref<TodoData[]>([])
-const searchParam = ref("")
-const holderArr = ref<TodoData[]>([])
-const showError = ref(false)
+  code: number;
+  data: TodoData[];
+};
+const isCompleteFilter = ref(true);
+const allData = ref<TodoData[]>([]);
+const searchParam = ref("");
+const showArrType = ref<"search" | "filter">("search");
+const searchArr = computed(() => {
+  return allData.value.filter((x) =>
+    x.todoName.toLowerCase().includes(searchParam.value.toLowerCase())
+  );
```

Comment:
I don't personally care much since you know the type of the inner variable in map/filter/reduce from the type of the array you're calling it on, but I know some of the other devs really prefer a more descriptive name for the variables in these array methods